dev/core#1664 - Fix errors recorded in log message on Manage Case scr…

jitendrapurohit opened the pull request:
…een and case dashboard page

Overview
----------------------------------------
Various errors recorded in log message on 

- Manage Case screen -> relationship tab.
- Case Dashboard Page.

Before
----------------------------------------
See [gitlab](https://lab.civicrm.org/dev/core/issues/1664) for steps to replicate the issue. 

In addition, visiting the case dashboard page records the below error in drupal logs-

>Notice: Undefined index: all in CRM_Case_Page_AJAX::getCases() (line 201 of /Users/jitendra/src/civicrm/CRM/Case/Page/AJAX.php).

After
----------------------------------------
No error message is recorded.

Comments
----------------------------------------
Gitlab - https://lab.civicrm.org/dev/core/issues/1664

demeritcowboy commented on CRM/Activity/Page/AJAX.php:
```diff
@@ -56,7 +56,7 @@ public static function getCaseGlobalRelationships() {
     CRM_Case_BAO_Case::getGlobalContacts($globalGroupInfo, NULL, FALSE, TRUE, NULL, NULL);
     // limit the rows
-    $relGlobal = CRM_Case_BAO_Case::getGlobalContacts($globalGroupInfo, $params['sortBy'], $showLinks = TRUE, FALSE, $params['offset'], $params['rp']);
+    $relGlobal = CRM_Case_BAO_Case::getGlobalContacts($globalGroupInfo, CRM_Utils_Array::value('sortBy', $params), $showLinks = TRUE, FALSE, $params['offset'], $params['rp']);
```
Thanks for the PR - I've noticed these messages too. FYI @colemanw might revoke your Civi card if he sees this line. Use `$params['sortBy'] ?? NULL`

jitendrapurohit replied:
Thanks @demeritcowboy. Have updated the PR now.

demeritcowboy commented on CRM/Case/Page/AJAX.php:
```diff
@@ -182,7 +182,7 @@ public static function getCases() {
     $params += CRM_Core_Page_AJAX::validateParams($requiredParameters, $optionalParameters);
 
-    $allCases = (bool) $params['all'];
+    $allCases = (bool) CRM_Utils_Array::value('all', $params, 0);
```
`$allCases = !empty($params['all']);`

jitendrapurohit replied:
done

demeritcowboy commented:
* General standards
    * [`r-explain`] __PASS__
    * [`r-user`] __PASS__
    * [`r-doc`] __PASS__
    * [`r-run`] __PASS__
      * Also checked the My Cases dashlet.
      * I didn't expect it to fix this, but just noting for other reviewers if you're testing sorting there is still a known separate sorting problem (https://lab.civicrm.org/dev/core/issues/1624)
* Developer standards
    * [`r-tech`] __PASS__
    * [`r-code`] __Issue__
      * The Array::value ones won't pass final review. They're so 2019.
    * [`r-maint`] __Undecided__
    * [`r-test`] __PASS__

eileenmcnaughton commented:
@jitendrapurohit - @colemanw  is on the warpath against CRM_Utils_Array::value - if you swap those out this is mergeable based on the review

jitendrapurohit commented:
ah, sorry, wasn't aware of that. Noticed loads of warning on case screen and quickly pushed a PR with the fix. I've updated the changes and removed the usage of Utils_Array now 👍

seamuslee001 commented:
Merging based on @demeritcowboy 's review
